src/helpers/query.ts

```diff
@@ -0,0 +1,15 @@
+import { PaginationParams } from '../types/request';
+import { isDefined, ValidateShape } from './typescript';
+
+export const getCollections = (collectionIds?: string[]) =>
+  isDefined(collectionIds) ? { collections: collectionIds.join() } : {};
+
+export const getFeedParams = <T>({
+  page,
+  perPage,
+  orderBy,
+}: ValidateShape<T, PaginationParams>) => ({
```

Comment:
The reason I added it here is to avoid a mistake that I only caught by luck. Example:

```ts
export type SearchParams = {
  query: string;
} & PaginationParams;

// no errros
createRequestParams((paginationParams: SearchParams) => ({
  pathname: `${SEARCH_PATH_PREFIX}/collections`,
  query: Query.getFeedParams(paginationParams),
}))

// should have been:
createRequestParams(({ query, ...paginationParams }: SearchParams) => ({
  pathname: `${SEARCH_PATH_PREFIX}/collections`,
  query: { query, ...Query.getFeedParams(paginationParams) },
}))
```

You're right that it should technically be everywhere but that's not plausible. I ended up deciding to keep it where I caught a bug.

Reply:
> I ended up deciding to keep it where I caught a bug.

I feel your pain about TS's lack of excess property checks in some scenarios, but using `ValidateShape` here just feels a bit random to me—other users might make the same mistake when they call other functions, and we have no protection there. For that reason, I'm not sure it's worth keeping this, unless we use it relentlessly.

Reply:
Fair, I'll remove it and swap it out with some tests for the API call arguments